### PR TITLE
Add list and restart tools views

### PR DESCRIPTION
**Pull request by andyhd**
# What

* Added tools routes and handlers for listing tools and restarting
* Added templates for listing tools

## How to review

0. Make sure you have a kubernetes config file at `$HOME/.kube/config`
1. Browse to http://localhost:3000/tools
2. See your RStudio instance listed (if you have one!)
3. Click on the `Restart` button
4. See that your RStudio instance is restarted (should get a 503 when trying to access it for a few seconds, or you can check with `kubectl get pods -n user-{your-username}`).

See [Trello #510](https://trello.com/c/wympY8SG) and [Trello #516](https://trello.com/c/EYuj7Gfs)

### CODE REVIEW

File: app/tools/routes.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const handlers = require('./handlers');
+
+
+module.exports = [
+  { name: 'list', pattern: '/tools', handler: handlers.list },
+  { name: 'restart', pattern: '/tools/:id/restart', handler: handlers.restart },
```
Comment (xoen) [inline app/tools/routes.js]: I would rename this param to `:name` as this is the tool name to restart, e.g. `rstudio`.

Comment (xoen) [reply app/tools/routes.js]: (I'm working on the endpoint to deploy a tool and the endpoint I have in mind is `POST /tools/:name/deploy`. It's nice to see we thought about similar endpoints)

File: app/k8s-api-client.js [hunk 2]
```diff
@@ -22,16 +22,16 @@ exports.get_namespace = get_namespace;
...
       throw new Error('Missing endpoint');
     }
 
-    let ns = namespace || this.namespace || 'default';
-
     const api = {
       'deployments': 'apis/apps/v1beta2',
       'pods': 'api/v1',
-    }[endpoint];
+    }[endpoint.split('/')[0]];
```
Comment (xoen) [inline app/k8s-api-client.js]: I probably didn't read the relevant code yet but...why the split here? I've seen both `Deployment` and `Pod` `endpoint()` returns a string without `/`, what am I missing here?

File: app/models.js [hunk 3]
```diff
@@ -141,13 +141,31 @@ class Deployment extends K8sModel {
...
   get_pods() {
     return Pod.list({ labelSelector: `app=${this.data.metadata.labels.app}` });
   }
 
   restart() {
     return Pod.delete_all({ labelSelector: `app=${this.data.metadata.labels.app}` });
   }
+
+  get_status() {
+    for (let i in this.status.conditions) {
```
Comment (xoen) [inline app/models.js]: Isn't this equivalent to:

```javascript
for (let condition in this.status.conditions) {
  // ...
}
```

File: app/tools/routes.js (hunk 1 above)
Comment (andyhd) [reply app/tools/routes.js]: Yep

File: app/k8s-api-client.js (hunk 2 above)
Comment (andyhd) [reply app/k8s-api-client.js]: This is a bugfix - the endpoint passed in can be something like `deployment/andyhd-rstudio-rstudio`, but that doesn't exist in the `api` lookup object here.

File: app/models.js (hunk 3 above)
Comment (andyhd) [reply app/models.js]: No, `for .. in` iterates through the indices (0, 1, 2, ...).
Perhaps you meant `for .. of`, which I didn't realise works with arrays! I will change it.

Comment (xoen) [reply app/models.js]: Yep, damn you JS :P

**Pull request merged**